### PR TITLE
Feat appwrite assistant

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

- Integrate assistant endpoint and container

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/config/services.php [hunk 1]
```diff
@@ -199,4 +199,17 @@
         'optional' => false,
         'icon' => '',
     ],
+    'assistant' => [
+        'key' => 'assistant',
+        'name' => 'Assistant',
+        'subtitle' => 'The Appwrite assistant service allows you to interact with appwrite assistant.',
```
Comment (eldadfux) [inline app/config/services.php:205]:
```suggestion
        'subtitle' => 'The Appwrite assistant service allows you to interact with Appwrite'\s AI assistant.',
```

Comment (lohanidamodar) [reply app/config/services.php:205]: Removing this for now, adding it to console service

File: app/controllers/api/console.php [hunk 2]
```diff
@@ -38,3 +39,57 @@
...
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_TEXT)
+    ->param('query', '', new Text(2000), 'Query')
+    ->inject('response')
+    ->action(function (string $query, Response $response) {
+        $ch = curl_init('http://appwrite-assistant:3003/');
+        $responseHeaders = [];
+        $query = json_encode(['prompt' => $query]);
+
+        $headers = ['accept: text/event-stream', 'authorization: ' . APP::getEnv('ASSISTANT_SECRET')];
```
Comment (eldadfux) [inline app/controllers/api/console.php:60]: Why do we need this secret? This container is going to be accessible only from private network

Comment (eldadfux) [reply app/controllers/api/console.php:60]: I think we can skip it, for the executor it was to protect different runtimes from reaching each other.

**Comment by eldadfux** [comment]
One more thing, we need to add abuse protection

File: app/controllers/api/console.php [hunk 3]
```diff
@@ -38,3 +39,59 @@
...
+    ->desc('Ask Query')
+    ->groups(['api', 'assistant'])
+    ->label('scope', 'public')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION])
+    ->label('sdk.namespace', 'assistant')
+    ->label('sdk.method', 'chat')
+    ->label('sdk.description', '/docs/references/assistant/chat.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_TEXT)
+    ->label('abuse-limit', 15)
```
Comment (eldadfux) [inline app/controllers/api/console.php:54]: Is the default time interval 1 hour?

**Review by eldadfux — APPROVED** [review]

File: app/controllers/api/console.php (hunk 3 above)
Comment (lohanidamodar) [reply app/controllers/api/console.php:54]: yes https://github.com/appwrite/appwrite/blob/fcd4a69ad1b3db67c6d89f02c5f230c2038a4b35/app/controllers/shared/api.php#L121

**Review by eldadfux — APPROVED** [review]

**Pull request merged**